docs/source/usage/workflow.rst

```diff
@@ -0,0 +1,30 @@
+.. _workflow:
+
+Usual workflow
+==============
+
+Deferred operation mode
+-----------------------
+
+IO operations are in general not performed by the openPMD API immediately after calling the corresponding API function.
+Rather, operations are enqueued internally and performed at so-called *flush points*.
+A flush point is a point within an application's sequential control flow where the openPMD API must uphold the following guarantees:
+
+*   In write mode, any change made to a user buffer whose data shall be stored in a dataset up to the flush point must be found in the written dataset.
+*   In write mode, no change made to a user buffer whose data shall be stored in a dataset after the flush point must be found in the written dataset.
+*   In read mode, a buffer into which data from a dataset should be filled, must not be altered by the openPMD API before the flush point.
+*   In read mode, a buffer into which data from a dataset should be filled, must have been filled with the requested data after the flush point.
+
+In short: ``storeChunk()`` and ``loadChunk()`` operations must happen exactly at flush points.
```

Comment:
Maybe:
```suggestion
In short: operations requested by ``storeChunk()`` and ``loadChunk()`` must happen exactly at flush points.
```
?

Reply:
We thought for some time, that we would allow these operations in the API generally between `store/loadChunk` and the `flush()`, but as we discussed recently, we do no want to keep that (do you remember the discussion @franzpoeschel? We should have taken minutes.)

Reply:
> (do you remember the discussion @franzpoeschel? We should have taken minutes.)

Yes, I do. The result of it was that I would document the logic in this PR, so that discussion is the reason why I added this :)

> I think the main reason was that we return the spans in those operations and then fill them before flush from the user side (for write).

Also, the `openpmd-pipe` tool already assumes that `storeChunk` works this way, even without spans.